### PR TITLE
Better coverage calculation numbers (`float64` -> `int`) - added some tests

### PR DESCRIPTION
**Pull request by magnetikonline**
We were seeing quite a few cases where coverage was pretty much identical, but floating point remainders would mis-report the result.

This PR at it's ❤️  flips the coverage values to integers where:

```
int(float64(CODE_COVERED) / float64(CODE_TOTAL) * 10000)
```

So...

- `6363` = `63.63%`
- `3175` = `31.75%`
- `0` = `00.00%` (obviously!)
- `205` = `2.05%`

The idea being that working with integers will result in somewhat more stable numbers when doing base vs. head compares. Two decimal places on percentages should be more than sufficient too.

In addition:

- Added some tests for the cover profile reporter/parser code.
- Some tests for the reporting messages given back to the pull request.
- Updated GitHub workflow from a lint to a test.
- Slightly more durable regular expression for parsing the cover profile lines
- Comments and some minor reformatting around the traps.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -29,30 +29,36 @@ func main() {
 
 	rootName := getModulePackageName()
 
+	// write report header
 	var buf bytes.Buffer
```
Comment (danielwhite) [inline main.go:33]: One might argue this should be https://pkg.go.dev/strings#Builder given how it is used. :D

**Review by danielwhite — APPROVED** [review]
OMG. Been meaning to figure this out for ages. Thanks so much.

File: main.go (hunk 1 above)
Comment (magnetikonline) [reply main.go:33]: Oh geez - well spotted - I'll add this in now 👍

Comment (magnetikonline) [reply main.go]: Fixed in 0ef64bb 👍

**Pull request merged**